ayaneo-platform.c

```diff
@@ -422,73 +368,140 @@ static void ayaneo_led_mc_set(u8 group, u8 pos, u8 brightness)
                 return;
 }
 
-static void ayaneo_led_mc_release(void) {
+static void ayaneo_led_mc_legacy_release(void)
+{
         if (!lock_global_acpi_lock())
                 return;
 
         ec_write(AYANEO_LED_MODE_REG, AYANEO_LED_MODE_RELEASE);
 
         if (!unlock_global_acpi_lock())
                 return;
+}
+
+static void ayaneo_led_mc_legacy_hold(void)
+{
+        if (!lock_global_acpi_lock())
+                return;
 
+        ec_write(AYANEO_LED_MODE_REG, AYANEO_LED_MODE_HOLD);
+
+        if (!unlock_global_acpi_lock())
+                return;
 }
 
-static void ayaneo_led_mc_intensity(u8 *color, u8 group, u8 zones[])
+static void ayaneo_led_mc_legacy_intensity(u8 group, u8 *color, u8 zones[])
 {
         int zone;
 
         for (zone = 0; zone < 4; zone++) {
-                ayaneo_led_mc_set(group, zones[zone], color[0]);
-                ayaneo_led_mc_set(group, zones[zone] + 1, color[1]);
-                ayaneo_led_mc_set(group, zones[zone] + 2, color[2]);
+                ayaneo_led_mc_legacy_set(group, zones[zone], color[0]);
+                ayaneo_led_mc_legacy_set(group, zones[zone] + 1, color[1]);
+                ayaneo_led_mc_legacy_set(group, zones[zone] + 2, color[2]);
         }
+        ayaneo_led_mc_legacy_set(0x03, 0x00, 0x00);
+}
+
```

Comment:
Could you call this in the loop for ayaneo_led_mc_legacy_intensity?

Reply:
Yeah, I've refactored that now, and added defines for a number of the commands as well to make their usage clearer.